packages/react-components/react-card/src/components/Card/useCardStyles.ts

```diff
@@ -189,30 +239,46 @@ export const useCardStyles_unstable = (state: CardState): CardState => {
     large: styles.sizeLarge,
   } as const;
 
-  const interactive =
-    state.root.onClick ||
-    state.root.onMouseUp ||
-    state.root.onMouseDown ||
-    state.root.onPointerUp ||
-    state.root.onPointerDown ||
-    state.root.onTouchStart ||
-    state.root.onTouchEnd;
+  const nonInteractiveAppearanceLookup = {
+    filled: styles.filled,
+    'filled-alternative': styles.filledAlternative,
+    outline: styles.outline,
+    subtle: styles.subtle,
+  } as const;
+
+  const interactiveAppearanceLookup = {
+    filled: styles.filledInteractive,
+    'filled-alternative': styles.filledAlternativeInteractive,
+    outline: styles.outlineInteractive,
+    subtle: styles.subtleInteractive,
+  } as const;
+
+  const appearanceLookup = {
+    nonInteractive: nonInteractiveAppearanceLookup,
+    interactive: interactiveAppearanceLookup,
+  } as const;
+
+  const selectedLookup = {
+    filled: styles.filledInteractiveSelected,
+    'filled-alternative': styles.filledAlternativeInteractiveSelected,
+    outline: styles.outlineInteractiveSelected,
+    subtle: styles.subtleInteractiveSelected,
+  } as const;
+
+  const interactive = state.selectable ? 'interactive' : 'nonInteractive';
 
   state.root.className = mergeClasses(
     cardClassNames.root,
     styles.root,
     sizeMap[state.size],
-    state.appearance === 'filled' && styles.filled,
-    state.appearance === 'filled-alternative' && styles.filledAlternative,
-    state.appearance === 'outline' && styles.outline,
-    state.appearance === 'subtle' && styles.subtle,
-    interactive && state.appearance === 'filled' && styles.filledInteractive,
-    interactive && state.appearance === 'filled-alternative' && styles.filledAlternativeInteractive,
-    interactive && state.appearance === 'outline' && styles.outlineInteractive,
-    interactive && state.appearance === 'subtle' && styles.subtleInteractive,
-    interactive && state.appearance !== 'outline' && styles.interactiveNoOutline,
+    appearanceLookup[interactive][state.appearance],
```

Comment:
Is this going too deep into the rabbit hole?

Reply:
IMO this is much better then adding another N lines of logic